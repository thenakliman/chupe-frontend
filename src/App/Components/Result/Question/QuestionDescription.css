.question-summary-item {
  padding: 1rem 9rem 1rem 2rem;
  width: 18rem;
  display: flex;
  justify-content: space-between;
}

.question-summary {
  margin: 3rem 0 1rem 0;
  font-size: 2rem;
  width: 58rem;
}

.question-description {
  display: flex;
}

.question-description-container {
  border: 1px solid #bbb;
  flex-wrap: wrap;
  justify-content: space-between;
  width: 58rem;
  display: flex;
}