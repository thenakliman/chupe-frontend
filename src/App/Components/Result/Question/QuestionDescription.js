import React from 'react';

require('./QuestionDescription.css');

/** Component for Question description */
export class QuestionDescription extends React.Component {
  /** Returns component to be rendered.
   *
   *  @return {object} component to be rendered
   */
  render() {
    return (
      <div id='question-description' className='question-description-with-summary'>
        <div id='question-summary' className='question-summary'>
          #{this.props.id} {this.props.summary}
        </div>
        <div className='question-description-container'>
          <div id='question-owner' className='question-summary-item'>
            <span>Asked By:</span>
            <span id='question-summary-asked-by'>{this.props.owner}</span>
          </div>
          <div id='question-assigned-to' className='question-summary-item'>
            <span>Assigned To: </span>
            <span id='question-summary-assigned-to'>
              {this.props.assignedTo}
            </span>
          </div>
          <div id='question-status' className='question-summary-item'>
            <span>Status: </span>
            <span id='question-summary-status'>{this.props.status}</span>
          </div>
          <div id='question-priority'className='question-summary-item'>
            <span>Priority: </span>
            <span id='question-summary-priority'>{this.props.priority}</span>
          </div>
        </div>
      </div>
    );
  }
}
