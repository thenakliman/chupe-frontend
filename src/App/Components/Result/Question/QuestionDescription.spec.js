/* eslint-disable */
import {QuestionDescription} from './QuestionDescription';
import React from 'react';
/* eslint-enable */
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';


describe('Question description', () => {
    it('should match the snapshot', () => {
        const wrapper = shallow(
            <QuestionDescription
              id={2}
              summary='some summary'
              owner='owner-2'
              assignedTo='assigned-to'
              status='my-status'
              priority='some priority'
              />);

        expect(toJson(wrapper)).toMatchSnapshot();
    });

  it('should have a question description div', () => {
      const wrapper = shallow(<QuestionDescription />);
      expect(wrapper.find('#question-description').length).toBe(1);
  });

  it('should have a question summary div with given text', () => {
      const summary = 'question summary';
      const wrapper = shallow(<QuestionDescription summary = {summary}/>);
      expect(wrapper.find('#question-summary').length).toBe(1);
  });

  it('should have a question summary', () => {
      const summary = 'question summary';
      const wrapper = shallow(<QuestionDescription summary = {summary}
                                                   id={10}/>);
      expect(wrapper.find('#question-summary').props().children
        ).toEqual(['#', 10, ' ', summary]);
  });

  it('should show question owner', () => {
      const owner = 'fake user';
      const wrapper = shallow(<QuestionDescription owner = {owner}/>);
      expect(wrapper.find('#question-summary-asked-by').props().children
          ).toEqual(owner);
  });

  it('should show question assigned to', () => {
      const assignedTo = 'fake user';
      const wrapper = shallow(<QuestionDescription assignedTo = {assignedTo}/>);
      expect(wrapper.find('#question-summary-assigned-to').props().children
          ).toEqual(assignedTo);
  });

  it('should show question status', () => {
      const status = 'OPEN';
      const wrapper = shallow(<QuestionDescription status = {status}/>);
      expect(wrapper.find('#question-summary-status').props().children
          ).toEqual(status);
  });

  it('should show question priority', () => {
      const priority = 'LOW';
      const wrapper = shallow(<QuestionDescription priority = {priority}/>);
      expect(wrapper.find('#question-summary-priority').props().children
          ).toEqual(priority);
  });
});
