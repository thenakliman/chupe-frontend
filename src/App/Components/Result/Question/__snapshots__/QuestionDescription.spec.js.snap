// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question description should match the snapshot 1`] = `
<div
  id="question-description"
>
  <div
    className="question-summary"
    id="question-summary"
  >
    #
    2
     
    some summary
  </div>
  <div
    className="question-description-container"
  >
    <div
      className="question-summary-item"
      id="question-owner"
    >
      <span>
        Asked By:
      </span>
      <span
        id="question-summary-asked-by"
      >
        owner-2
      </span>
    </div>
    <div
      className="question-summary-item"
      id="question-assigned-to"
    >
      <span>
        Assigned To: 
      </span>
      <span
        id="question-summary-assigned-to"
      >
        assigned-to
      </span>
    </div>
    <div
      className="question-summary-item"
      id="question-status"
    >
      <span>
        Status: 
      </span>
      <span
        id="question-summary-status"
      >
        my-status
      </span>
    </div>
    <div
      className="question-summary-item"
      id="question-priority"
    >
      <span>
        Priority: 
      </span>
      <span
        id="question-summary-priority"
      >
        some priority
      </span>
    </div>
  </div>
</div>
`;
